src/components/input/select/index.tsx

```diff
@@ -80,43 +82,43 @@ export class SmoothlyInputSelect implements Input, Editable, Clearable, Componen
 	}
 	@Method()
 	async reset(): Promise<void> {
+		this.selected.forEach(item => (item.selected = item.hidden = false))
+		this.initialValue.forEach(item => (item.selected = true))
+		this.selected = [...this.initialValue]
+		this.displaySelected()
 		this.changed = false
-		this.selectedElement && (this.selectedElement.hidden = false)
-		this.mainElement &&
-			(this.mainElement.innerHTML =
-				this.initialValue !== undefined ? this.initialValue.innerHTML : this.placeholder ? this.placeholder : "")
-		this.selectedElement = this.initialValue
-		this.selectedElement && !this.showSelected && (this.selectedElement.hidden = true)
 	}
 
 	@Method()
 	async clear(): Promise<void> {
 		if (this.clearable) {
-			this.selectedElement = undefined
+			this.selected.forEach(item => (item.selected = item.hidden = false))
+			this.selected = []
 			if (this.mainElement) {
 				this.mainElement.innerHTML = this.placeholder ?? "(none)"
 			}
 		}
 	}
-
 	@Method()
 	async edit(editable: boolean): Promise<void> {
 		this.readonly = !editable
 	}
-
 	@Method()
 	async setInitialValue(): Promise<void> {
 		this.changed = false
-		this.initialValue = this.selectedElement
-		this.smoothlyInput.emit({ [this.name]: this.selectedElement?.value })
+		this.initialValue = [...this.selected]
 	}
-	@Watch("selectedElement")
-	onSelectedChange(value: HTMLSmoothlyItemElement | undefined, old: HTMLSmoothlyItemElement | undefined): void {
-		this.initialValueHandled && (this.changed = this.initialValue !== this.selectedElement)
-		if (old)
-			old.selected = false
-		this.smoothlySelect.emit(value?.value)
-		this.smoothlyInput.emit({ [this.name]: value?.value })
+	@Watch("selected")
+	onSelectedChange() {
+		this.initialValueHandled && (this.changed = !this.areValuesEqual(this.selected, this.initialValue))
```

Comment:
im looking and i can't see what the "changed" Prop is meant to do, can we remove it?

Reply:
If you place a button (in the demo there is a reset button for example) inside the select it will only be displayed in case changed is true

Reply:
ok make it state, then

Reply:
Cant do that because it needs to forfill Interface so that it will work in form